src/bit-systems/audio-zone-system.ts

```diff
@@ -0,0 +1,325 @@
+import { defineQuery, enterQuery, exitQuery } from "bitecs";
+import { getScene, HubsWorld } from "../app";
+import { AudioEmitter, AudioZone } from "../bit-components";
+import { Box3, BoxGeometry, DoubleSide, MeshBasicMaterial, Object3D, Ray, Vector3, Mesh, BoxHelper } from "three";
+import { AUDIO_ZONE_FLAGS } from "../inflators/audio-zone";
+import { disposeMaterial, disposeNode } from "../utils/three-utils";
+import { AudioSettings } from "../components/audio-params";
+import { updateAudioSettings } from "../update-audio-settings";
+
+const debugObjects = new Map<number, Object3D>();
+
+const DEBUG_BBAA_COLOR = 0x49ef4;
+
+let debugMaterial: MeshBasicMaterial | undefined | null;
+
+const createMaterial = () => {
+  if (!debugMaterial) {
+    debugMaterial = new MeshBasicMaterial({
+      color: DEBUG_BBAA_COLOR,
+      transparent: true,
+      opacity: 0.25,
+      side: DoubleSide
+    });
+  }
+};
+
+getScene().then(() => {
```

Comment:
I don't think it's correct to use `getScene` as a hook like this. Strong evidence (to me) is that you aren't _using_ the scene returned by `getScene`. (`getScene` is already a hack that we should find a way to remove -- We know when the scene exists, so if there's work that needs to happen after the scene exists, then we should schedule the work then.) 



If you need to do some async work before rendering begins, you can use `preload`. But this doesn't seem to be async. It seems like it can happen as soon as `world` and `store` exist. If that's the case, you can wrap this in a function:

```ts
const audioZoneQuery = defineQuery([AudioZone]);
export function maybeToggleDebugPanel(world: HubsWorld, store: Store) {
    const isDebugEnabled = store.state.preferences.showAudioDebugPanel;
    if (isDebugEnabled) createMaterial();
    audioZoneQuery(APP.world).forEach(zoneEid => {
      if (isDebugEnabled) {
        !debugObjects.has(zoneEid) && addZoneDebugObject(APP.world, zoneEid);
      } else {
        debugObjects.has(zoneEid) && releaseZoneDebugObject(APP.world, zoneEid);
      }
    });
    if (!isDebugEnabled && debugMaterial) {
      disposeMaterial(debugMaterial);
      debugMaterial = null;
    }
}
```
Then we can call this function whenever we need to (e.g. in hub.js or app.tsx, where you have non-global references to both store and world)
```ts
// As soon as we have store and world...
store.addEventListener("statechanged", () => {
  maybeToggleDebugPanel(world, store);
);
maybeToggleDebugPanel(world, store);
```

Reply:
I agree that this is not ideal and what you mention seems reasonable. The only thing that I don't like about that is that it forces the system developer to spread system code around the codebase. I'd like to discuss more the idea that I shared in my previous comment here: https://github.com/mozilla/hubs/pull/5944#discussion_r1109761735

i think I'm going to leave it as is for now and add a TODO to revisit this when we have discussed a better path for preferences updates.